Scripts/SL-FavoritesHandler.lua

```diff
@@ -0,0 +1,96 @@
+-- SM5 Favorites manager by leadbman & modified for RIO by Rhythm Lunatic
+-- Modified and implemented by Crash Cringle
+
+-- Inhibit Regular Expression magic characters ^$()%.[]*+-?)
+local function strPlainText(strText)
+	-- Prefix every non-alphanumeric character (%W) with a % escape character,
+	-- where %% is the % escape, and %1 is original character
+	return strText:gsub("(%W)","%%%1")
+end
+
+getFavoritesPath = function(player)
+	local path = PROFILEMAN:GetProfileDir(ProfileSlot[PlayerNumber:Reverse()[player]+1]).."FavoriteSongs.txt";
```

Comment:
I might want to keep this as "favorites.txt" instead of "FavoriteSongs.txt" since that's how the current release has it. Also might be worth checking the old path in "Other/SongManager playername.txt" if the player profile favorites file doesn't exist.